agent/testflinger_agent/agent.py

```diff
@@ -202,11 +205,17 @@ def process_jobs(self):
         self.retry_old_results()
 
         self.check_restart()
-
+        event_emitter = EventEmitter(None, None, self.client)
```

Comment:
This is created too early and could leak into the next job. Delay creation of this event_emitter until we actually start processing a job.